packages/modules/pricing/src/repositories/pricing.ts

```diff
@@ -53,28 +52,49 @@ export class PricingRepository
     }
 
     // Generate flatten key-value pairs for rule matching
-    const [ruleAttributes, priceListRuleAttributes] =
-      await this.getAttributesFromRuleTables(knex)
-
-    const allowedRuleAttributes = [
-      ...ruleAttributes,
-      ...priceListRuleAttributes,
-    ]
-
     const flattenedKeyValuePairs = flattenObjectToKeyValuePairs(context)
 
-    const flattenedContext = Object.entries(flattenedKeyValuePairs).filter(
-      ([key, value]) => {
+    // First filter by value presence
+    const candidateContext = Object.entries(flattenedKeyValuePairs).filter(
+      ([, value]) => {
         const isValuePresent = !Array.isArray(value) && isPresent(value)
         const isArrayPresent = Array.isArray(value) && value.flat(1).length
 
-        return (
-          allowedRuleAttributes.includes(key) &&
-          (isValuePresent || isArrayPresent)
-        )
+        return isValuePresent || isArrayPresent
       }
     )
 
+    // If we have potential context keys, efficiently check which ones exist in rules
+    const flattenedContext = candidateContext
+    // if (candidateContext.length > 0) {
+    //   const contextKeys = candidateContext.map(([key]) => key)
+
+    //   // Efficient query to check if keys exist in either rule table
+    //   // This avoids fetching ALL rule attributes - only checks our specific keys
+    //   const existingKeys = await knex
+    //     .raw(
+    //       `
+    //     SELECT DISTINCT attribute
+    //     FROM (
+    //       SELECT attribute FROM price_rule WHERE attribute IN (${contextKeys
+    //         .map(() => "?")
+    //         .join(",")})
+    //       UNION
+    //       SELECT attribute FROM price_list_rule WHERE attribute IN (${contextKeys
+    //         .map(() => "?")
+    //         .join(",")})
+    //     ) AS combined_attributes
+    //   `,
+    //       [...contextKeys, ...contextKeys]
+    //     )
+    //     .then((result) => result.rows.map((row) => row.attribute))
+
+    //   // Filter context to only include keys that exist in rules
+    //   flattenedContext = candidateContext.filter(([key]) =>
+    //     existingKeys.includes(key)
+    //   )
+    // }
```

Comment:
```suggestion
const flattenedContext = candidateContext
```

This is a nice alternative if the test doesn't pass! Remove if it does 🤞🏻

Reply:
let's jsut get this removed and it's perfect.

Reply:
I ve fixed the flatten util now it should be good to remove and I did just that 😓